elastic/kibana
# [Serverless][Advanced Settings] Add validations on user input

Addresses https://github.com/elastic/kibana/issues/168684

## Summary

This PR adds validations of the user input for a some of the uiSettings (see https://github.com/elastic/kibana/issues/168684) based on the `schema` that is provided in the setting definitions. We do this by exposing a `validateValue` API from the uiSettings browser-side client, which calls an internal validate route to the server. Then `validateValue` is called in the `onChange` handlers of the setting input fields, utilising  `debounce` to reduce the number of requests to the server.

Note: One of the validation requirements is for validating that the `defaultIndex` setting value is an existing data view. This is a more complicated validation and will be addressed in a separate PR.

![defaultAnomalyScore](https://github.com/elastic/kibana/assets/59341489/4fb99993-9c2d-4c92-8ea4-bacdb58f7be4)
![defaultThreatIndex](https://github.com/elastic/kibana/assets/59341489/264fa663-43cb-44a1-9435-4de0fa1e957a)


### How to test
**Testing the validations in serverless:**
1. Start Es with `yarn es serverless` and Kibana with `yarn serverless-{es/oblt/security}`
2. Go to Management -> Advanced settings
3. Verify that the UI provides correct validation error messages for [the settings that require validation](https://github.com/elastic/kibana/issues/168684).

**Verify that self-managed Kibana is not affected:**
There should be no validations in self-managed Kibana and setting an invalid value for an advanced setting should be allowed and shouldn't cause any errors.

@elastic/kibana-core team, I know you're no longer code owners of the uiSettings service, but any feedback on those changes will be highly appreciated. 🙏
